crates/plugins/src/badger/mod.rs

```diff
@@ -0,0 +1,318 @@
+        })
+    }
+
+    fn should_check(&self) -> bool {
+        match self.previous_badger {
+            PreviousBadger::Fresh => true,
+            PreviousBadger::FromCurrent { when, .. } => has_timeout_expired(when),
+        }
+    }
+
+    fn fire_and_forget_update() {
+        if let Err(e) = Self::fire_and_forget_update_impl() {
+            tracing::info!("Failed to launch plugins update process; checking using latest local repo anyway. Error: {e:#}");
+        }
+    }
+
+    fn fire_and_forget_update_impl() -> anyhow::Result<()> {
+        let mut update_cmd = tokio::process::Command::new(std::env::current_exe()?);
+        update_cmd.args(["plugins", "update"]);
+        update_cmd.stdout(std::process::Stdio::null());
+        update_cmd.stderr(std::process::Stdio::null());
+        update_cmd.spawn()?;
+        Ok(())
+    }
+
+    async fn check(&self) -> anyhow::Result<BadgerUI> {
+        let available_upgrades = self.available_upgrades().await?;
+
+        // TO CONSIDER: skipping this check and badgering for the same upgrade in case they missed it
+        if self
+            .previous_badger
+            .includes_any(&available_upgrades.list())
+        {
+            return Ok(BadgerUI::None);
+        }
+
+        if !available_upgrades.is_none() {
+            self.record_manager
+                .record_badger(
+                    &self.plugin_name,
+                    &self.current_version,
+                    &available_upgrades.list(),
+                )
+                .await
+        };
+
+        Ok(available_upgrades.classify())
+    }
+
+    async fn available_upgrades(&self) -> anyhow::Result<AvailableUpgrades> {
+        let store = self.plugin_manager.store();
+
+        let latest_version = {
+            let latest_lookup = crate::lookup::PluginLookup::new(&self.plugin_name, None);
+            let latest_manifest = latest_lookup
+                .get_manifest_from_repository(store.get_plugins_directory())
+                .await
+                .ok();
+            latest_manifest.and_then(|m| semver::Version::parse(m.version()).ok())
+        };
+
+        let manifests = store.catalogue_manifests()?;
+        let relevant_manifests = manifests
+            .into_iter()
+            .filter(|m| m.name() == self.plugin_name);
+        let compatible_manifests = relevant_manifests.filter(|m| {
+            m.has_compatible_package() && m.is_compatible_spin_version(self.spin_version)
+        });
+        let compatible_plugin_versions =
+            compatible_manifests.filter_map(|m| PluginVersion::try_from(m, &latest_version));
+        let considerable_manifests = compatible_plugin_versions
+            .filter(|pv| !pv.is_prerelease() && pv.is_higher_than(&self.current_version))
+            .collect::<Vec<_>>();
+
+        let (eligible_manifests, questionable_manifests) = if self.current_version.major == 0 {
+            (vec![], considerable_manifests)
+        } else {
```

Comment:
Given that none of our plugins are v1.0, this badger will always throw "but may not be backward compatible with your current plugin", even if updating a patch version. For example, if i run `spin cloud --help` with `cloud 0.1.0` installed, I get:
```sh
This plugin can be upgraded. Version 0.1.1 is available,
but may not be backward compatible with your current plugin.
```
According to semver, it should be backwards compat

Reply:
I thought semver had an exception for major version 0?  If it doesn't, we've been doing it wrong... _grin_

Reply:
Indeed you are right: "
Major version zero (0.y.z) is for initial development. Anything MAY change at any time. The public API SHOULD NOT be considered stable."

I guess my point is, are we concerned with every badger saying "may not be compatible". I guess this is a given, but i do think for our plugins it would be worth visiting what it would take to see them to `1.0`

Reply:
Yes, and this perhaps comes back to the discussion we had about stability vs maturity - it would be great to try to get our plugins to 1.x _even if_ they are not fully mature, so that we can at least send meaningful signals about interface stability and compatibility.  But the risk then is we have to do too many major revs in the early months...